python/lsst/pex/config/config.py

```diff
@@ -1250,7 +1250,7 @@ def saveToStream(self, outfile, root="config", skipImports=False):
                 outfile.write(f"import {configType.__module__}\n")
                 outfile.write(f"assert type({root})=={typeString}, 'config is of type %s.%s instead of "
                               f"{typeString}' % (type({root}).__module__, type({root}).__name__)\n")
-                for imp in self._imports:
+                for imp in sorted(self._imports):
```

Comment:
Is this safe to do? imports sometimes have side effects, simply switching the ordering around may not be safe.

Reply:
To clarify, we can't consider all import side-effects bugs - after all, we're importing in configs _precisely_ because we register classes in various module-scope dictionaries as side effects - but I do think we should consider any import-time side effect that is not idempotent and independent of the order of imports a bug.  In practice, import-order fragility in our own code has only ever come (to my knowledge) from module dependency cycles, which I do think are best fixed.  Doing `from . import <lifted-symbol>` and the afw cross-subpackage pybind11 wrappers are the biggest causes of those, and I think the afw ones are actually now addressed by the `WrapperCollection` stuff in [cpp?]utils.